### PR TITLE
Add nativeFinish JNI for destruct game script env.

### PR DESCRIPTION
**Pull request by yangws**

### CODE REVIEW

File: cocos/platform/android/jni/JniImp.cpp [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #include <android/asset_manager_jni.h>
 #include <jni.h>
 #include <mutex>
+#include <audio/include/AudioEngine.h>
```
Comment (dumganhar) [inline cocos/platform/android/jni/JniImp.cpp]: Uses `#include "..."` instead of <>

File: cocos/platform/android/jni/JniImp.cpp [hunk 2]
```diff
@@ -144,8 +147,20 @@ extern "C"
         g_app->start();
     }
 
+    JNIEXPORT void Java_org_cocos2dx_lib_Cocos2dxRenderer_nativeDone(JNIEnv*  env, jobject thiz)
+    {
+        g_scriptEngineDone = true;
+        LOGD("CocosRenderer.nativeExit");
```
Comment (dumganhar) [inline cocos/platform/android/jni/JniImp.cpp]: init <-> destroy. So probably, nativeDestroy should be a better name.

Comment (yangws) [reply cocos/platform/android/jni/JniImp.cpp]: create<->destroy  init<->done

File: cocos/platform/android/jni/JniImp.cpp [hunk 3]
```diff
@@ -76,6 +77,7 @@ namespace
     int g_deviceAudioBufferSizeInFrames = 192;
     int g_width = 0;
     int g_height = 0;
+    bool g_gameFinished = false;
```
Comment (dumganhar) [inline cocos/platform/android/jni/JniImp.cpp]: -> g_isGameFinished

**Pull request merged**